src/Microsoft.OData.Core/JsonLight/ODataJsonLightContextUriParser.cs

```diff
@@ -71,19 +71,44 @@ private ODataJsonLightContextUriParser(IEdmModel model, Uri contextUriFromPayloa
             ODataPayloadKind payloadKind,
             Func<IEdmType, string, IEdmType> clientCustomTypeResolver,
             bool needParseFragment,
-            bool throwIfMetadataConflict = true)
+            bool throwIfMetadataConflict = true,
+            Uri baseUri = null,
+            string navigationSource = null)
         {
             if (contextUriFromPayload == null)
             {
                 throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_NullMetadataDocumentUri);
             }
 
-            // Create an absolute URI from the payload string
-            // TODO: Support relative context uri and resolving other relative uris
+            // Create a context URI from the payload string.
             Uri contextUri;
             if (!Uri.TryCreate(contextUriFromPayload, UriKind.Absolute, out contextUri))
             {
-                throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_TopLevelContextUrlShouldBeAbsolute(contextUriFromPayload));
+                if (baseUri != null)
+                {
+                    ODataUri oDataUri = new ODataUri() { ServiceRoot = baseUri };
+
+                    // Graph requires this format for delta request body context.
+                    if (contextUriFromPayload.Equals("#$delta", StringComparison.OrdinalIgnoreCase))

```

Comment:
Don't we have a constant for `$delta`? Can we use it here?

Reply:
the contextUriFromPayload will be `#$delta`..This means we'll have to do some concatenations before doing the check.

Reply:
fixed